File: Python/Problems/Problems/problem2798.py
from typing import List

from .problem import Problem


class Problem2798(Problem):
    def test(self) -> bool:
        hours = [5, 1, 4, 2, 2]
        target = 6

        expected = 0

        result = self.numberOfEmployeesWhoMetTarget(hours, target)

        print(result)

        return expected == result

    def numberOfEmployeesWhoMetTarget(self, hours: List[int], target: int) -> int:
        result = 0

        for hour in hours:
            if hour >= target:
                result += 1

        return result

File: C++/Problems/Problems/problem_2798.cpp
#include "problem_2798.h"

bool problem_2798::test() {
    std::vector hours { 5, 1, 4, 2, 2 };
    int target = 6;

    int expected = 0;

    auto result = numberOfEmployeesWhoMetTarget(hours, target);

    std::cout << result << std::endl;

    return expected == result;
}

int problem_2798::numberOfEmployeesWhoMetTarget(std::vector<int> &hours, int target) {
    int result = 0;

    for (auto hour : hours)
        if (hour >= target)
            result++;

    return result;
}
